mission/functions/core/teams/fn_force_team_change.sqf

```diff
@@ -50,9 +50,10 @@ _nextPlayerTeamArray pushBackUnique _player;
 
 [[_team], {
 	[] call vn_mf_fnc_task_refresh_tasks_client;
-	[] call vn_mf_fnc_tr_overview_team_update;
 	[] call vn_mf_fnc_update_channels;
 	[] call vn_mf_fnc_apply_unit_traits;
 	[] call vn_mf_fnc_action_trait;
-	// reset the available actions on the client
+	// DO THIS LAST
+	// so the roles update properly in the teams info pages of the taskroster UI
+	[] call vn_mf_fnc_tr_overview_team_update;
```

Comment:
roles need to be updated by calling `vn_mf_fnc_apply_unit_traits` before we update the task roster display in `vn_mf_fnc_tr_overview_team_update`

Reply:
!todo: add a comment line `// otherwise when a player changes teams their roles are not updated until they reopen the task roster`